promql/parse_test.go

```diff
@@ -122,39 +124,77 @@ var testExpr = []struct {
 			},
 		},
 	}, {
-		input: "", fail: true,
+		input:  "",
+		fail:   true,
+		errMsg: "no expression found in input",
+	}, {
+		input:  "# just a comment\n\n",
+		fail:   true,
+		errMsg: "no expression found in input",
 	}, {
-		input: "# just a comment\n\n", fail: true,
+		input:  "1+",
+		fail:   true,
+		errMsg: "could not find right-hand side of binary expression",
 	}, {
-		input: "1+", fail: true,
+		input:  ".",
+		fail:   true,
+		errMsg: "unrecognized character in statement: '.'",
```

Comment:
Ok, then maybe just "character" instead of "character in statement", if it can happen both in a statement and an expression?